### PR TITLE
[FUNK-1550] IP Allowlisting Changes

### PR DESCRIPTION
**Pull request by vaibhav-nanda**
### Proposed changes

https://segment.atlassian.net/browse/FUNK-1550
The changes are to handle the Functions IP Allowlisting changes made.

### CODE REVIEW

File: src/connections/functions/index.md [hunk 1]
```diff
@@ -46,4 +46,10 @@ To learn more, visit [destination insert functions](/docs/connections/functions/
 
 With Functions Copilot, you can instrument custom integrations, enrich and transform data, and even secure sensitive data nearly instantaneously without writing a line of code. 
 
-To learn more, visit the [Functions Copilot documentation](/docs/connections/functions/copilot/).
\ No newline at end of file
+To learn more, visit the [Functions Copilot documentation](/docs/connections/functions/copilot/).
+
+#### IP Allowlisting for functions
+
+With IP Allowlisting, users can ensure that outbound traffic from Functions will originate only from a predefined set of IP addresses. This provides added security for external services by making it easier to recognize and allowlist trusted traffic sources.
```
Comment (forstisabella) [inline src/connections/functions/index.md:53]:
```suggestion

> warning " "
> IP Allowlisting is only available for customers on Business Tier plans.

IP Allowlisting uses a NAT gateway to route outbound Functions traffic from Segment’s servers to your destinations through a limited range of IP addresses, which can prevent malicious actors from establishing TCP and UDP connections with your integrations.
```

File: src/connections/functions/index.md [hunk 2]
```diff
@@ -46,4 +46,10 @@ To learn more, visit [destination insert functions](/docs/connections/functions/
 
 With Functions Copilot, you can instrument custom integrations, enrich and transform data, and even secure sensitive data nearly instantaneously without writing a line of code. 
 
-To learn more, visit the [Functions Copilot documentation](/docs/connections/functions/copilot/).
\ No newline at end of file
+To learn more, visit the [Functions Copilot documentation](/docs/connections/functions/copilot/).
+
+#### IP Allowlisting for functions
```
Comment (forstisabella) [inline src/connections/functions/index.md:51]:
```suggestion
#### IP Allowlisting
```

File: src/connections/functions/index.md [hunk 3]
```diff
@@ -46,4 +46,10 @@ To learn more, visit [destination insert functions](/docs/connections/functions/
...
+
+With IP Allowlisting, users can ensure that outbound traffic from Functions will originate only from a predefined set of IP addresses. This provides added security for external services by making it easier to recognize and allowlist trusted traffic sources.
+
+To learn more, visit [IP Allowlisting documentation](/docs/connections/destinations/#ip-allowlisting).
```
Comment (forstisabella) [inline src/connections/functions/index.md:55]: Should we remove the note in the Destination Overview page that mentions Destination Functions are not supported?

```suggestion
To learn more, visit [Segment's IP Allowlisting documentation](/docs/connections/destinations/#ip-allowlisting).
```

**Review by forstisabella — DISMISSED** [review]

File: src/connections/functions/index.md (hunk 1 above)
Comment (vaibhav-nanda) [reply src/connections/functions/index.md]: The warning `IP Allowlisting is only available for customers on Business Tier plans` has been added as a simple text as it is being displayed as  simple text in Destinations IP Allowlisting doc as well.
Ref: https://segment.com/docs/connections/destinations/#ip-allowlisting

**Review by forstisabella — APPROVED** [review]

**Pull request merged**